src/dbus_stats.rs

```diff
@@ -0,0 +1,342 @@
+    });
+}
+
+fn parse_peer_accounting(owned_value: &OwnedValue) -> Option<HashMap<String, DBusBrokerPeerAccounting>> {
+    let value: &Value = owned_value;
+    let peers_value = match value {
+        Value::Array(peers_value) => peers_value,
+        _ => return None,
+    };
+
+    let mut result = HashMap::new();
+    for peer_value in peers_value.iter() {
+        if let Some(peer) = parse_peer_struct(peer_value) {
+            result.insert(peer.name.clone(), peer);
+        }
+    }
+
+    Some(result)
+}
+
+/* Parse DBusBrokerUserAccounting from OwnedValue.
+ * Expected structure:
+ * struct {
+ *     uint32 0
+ *     array [
+ *         struct {
+ *             string "Bytes"
+ *             uint32 536843240
+ *             uint32 536870912
+ *         }
+ *         ... more fields
+ *     ]
+ *     # TODO parse usages, ignoring for now
+ *     # see src/bus/driver.c:2258
+ *     # the part below is not parsed
+ *     array [
+ *         dict entry(
+ *             uint32 0
+ *             array [
+ *             dict entry(
+ *                 string "Bytes"
+ *                 uint32 27672
+ *             )
+ *             ... more fields
+ *             ]
+ *         )
+ *     ]
+ * }
+ */
+
+fn parse_user_struct(user_value: &Value) -> Option<DBusBrokerUserAccounting> {
+    let user_struct = match user_value {
+        Value::Structure(user_struct) => user_struct,
+        _ => return None,
+    };
+
+    let fields = user_struct.fields();
+    if fields.len() < 2 {
+        return None;
+    }
+
+    // 1. Extract name (Str)
+    let uid = match &fields[0] {
+        Value::U32(n) => *n,
+        _ => return None,
+    };
+
+    // 2. Extract user stats
+    let user_stats = match &fields[1] {
+        Value::Array(a) => a,
+        _ => return None,
+    };
+
+    let mut user = DBusBrokerUserAccounting::default();
+    user.uid = uid;
+
+    for user_stat in user_stats.iter() {
+        if let Value::Structure(user_stat) = user_stat {
+            let field_tuple = user_stat.fields();
+            if field_tuple.len() < 3 {
+                continue;
+            }
+
+            let name = match &field_tuple[0] {
+                Value::Str(s) => s.to_string(),
+                _ => continue,
+            };
+
+            let pair = CurMaxPair{
+                cur: match &field_tuple[1] {
+                    Value::U32(n) => *n,
+                    _ => continue,
+                },
+                max: match &field_tuple[2] {
+                    Value::U32(n) => *n,
+                    _ => continue,
+                },
+            };
+
+            match name.as_str() {
+                "Bytes" => user.bytes = Some(pair),
+                "Fds" => user.fds = Some(pair),
+                "Matches" => user.matches = Some(pair),
+                "Objects" => user.objects = Some(pair),
+                _ => {} // ignore other fields
+            }
+        }
+    }
+
+    return Some(user);
```

Comment:
```suggestion
    Some(user)
```
Implicit return - Notice the lack of `;` here ...